etcdmain/config.go

```diff
@@ -283,6 +283,11 @@ func (cfg *config) parse(arguments []string) error {
 	}
 
 	var err error
+
```

Comment:
Maybe add a comment mentioning that we have to treat this env variable differently than the other env variables, because we need to determine whether to use or ignore the env variables based on if the config file is set.